internal/plugin/connectors/tcp/mssql/connector.go

```diff
@@ -186,32 +201,40 @@ func (connector *SingleUseConnector) Connect(
 	// Write the prelogin packet back to the user
 	err = mssql.WritePreloginWithPacketType(clientBuffer, preloginResponse, mssql.PackReply)
 	if err != nil {
-		return connector.sendError("failed to write prelogin response: %s", err)
+		wrappedError := errors.Wrap(err, "failed to write prelogin response")
+		connector.sendError(wrappedError)
+		return nil, wrappedError
 	}
 
-	// Just like above, we don't need to concern ourselves with what the client would
-	// typically be sending to the server, since we are handling all communication with
-	// the server ourselves. We read the next packet so the client isn't blocked,
-	// while the driver handles the handshake.
-	err = clientBuffer.ReadNextPacket()
+	// We parse the client's Login packet so that we can pass on params to the server.
+	clientLogin, err := mssql.ReadLogin(clientBuffer)
```

Comment:
Just a thought that's not directly tied to your change here, sparked by a comment that you made last week on whether our approach for the mssql connector is too prescriptive. We're assuming everything between client and server falls in lock-step, but I wonder if we'll recover (eventually) if one side or the other gets out of step.

Reply:
I think timeouts resolve this. There's a built in timeout.